modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/util/RelayUtils.java

```diff
@@ -247,24 +248,26 @@ public static boolean isDeleteRequestWithoutPayload (MessageContext msgContext)
      *
      */
     public static boolean shouldOverwriteContentType(MessageContext msgContext, TargetRequest request) {
-
         boolean builderInvoked = Boolean.TRUE.equals(msgContext
                 .getProperty(PassThroughConstants.MESSAGE_BUILDER_INVOKED));
-
         boolean noEntityBodySet =
                 Boolean.TRUE.equals(msgContext.getProperty(PassThroughConstants.NO_ENTITY_BODY));
-
         Map<String, LinkedHashSet<String>> headers = request.getHeaders();
         boolean contentTypeInRequest = false;
         if (headers.size() != 0 && (headers.get("Content-Type") != null || headers.get("content-type") != null)) {
             contentTypeInRequest = true;
         }
-        
-        // If builder is not invoked, which means the passthrough scenario, we should overwrite the content-type 
+        boolean isDefaultContentTypeEnabled = false;
+        ConfigurationContext configurationContex = msgContext.getConfigurationContext();
+        if (configurationContex != null && configurationContex.getAxisConfiguration()
+                .getParameter(NhttpConstants.REQUEST_CONTENT_TYPE) != null) {
+            isDefaultContentTypeEnabled = true;
```

Comment:
What will be the behavior if the defaultContentType is **empty/content** ?

Reply:
It will still use that content type. Is there some other expected behavior for it?

Reply:
Will it be written to the output request with this fix

Reply:
this fix only checks whether that property is added and if its added it will return true,  setting the content type is already done here 

https://github.com/wso2-support/wso2-synapse/blob/support-2.1.7-wso2v65/modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/util/TargetRequestFactory.java#L115

Reply:
> this fix only checks whether that property is added and if its added it will return true, setting the content type is already done here

https://github.com/wso2-support/wso2-synapse/blob/support-2.1.7-wso2v65/modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/util/TargetRequestFactory.java#L115

yes true. but based on this only result only we over write the header.

> It will still use that content type. Is there some other expected behavior for it?

If is is empty/content , we shouldn't overwrite the header with that value and we need to skip that in both response and request. Can you please verify whether it is getting skipped there ?

Also better if we check REQUEST_CONTENT_TYPE is not empty in addition to null check.

Reply:
if it is empty/content it does not override the header in this case, we only check if the header is null. This is done in the request path.